dioptas/widgets/plot_widgets/NormalizedImageItem.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+# Dioptas - GUI program for fast processing of 2D X-ray diffraction data
+# Principal author: Clemens Prescher (clemens.prescher@gmail.com)
+# Copyright (C) 2014-2019 GSECARS, University of Chicago, USA
+# Copyright (C) 2015-2018 Institute for Geology and Mineralogy, University of Cologne, Germany
+# Copyright (C) 2019-2020 DESY, Hamburg, Germany
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+
+from contextlib import contextmanager
+from typing import Optional
+
+import numpy as np
+import pyqtgraph as pg
+
+
+class Normalization:
+    """Class defining the interface for implementing normalizations"""
+
+    ID = ""
+    """Abbreviation name of the normalization, must be unique among Normalization"""
+
+    SHORTNAME = ""
+    """Short name or symbol to use as e.g., button text"""
+
+    DESCRIPTION = ""
+    """Human-readable name of the normalization"""
+
+    @staticmethod
+    def apply(data: np.ndarray) -> np.ndarray:
+        """Forward conversion of data to normalized data"""
+        raise NotImplementedError()
+
+    @staticmethod
+    def revert(data: np.ndarray) -> np.ndarray:
+        """Backward conversion of normalized data to data"""
+        raise NotImplementedError()
+
+    @staticmethod
+    def invalid(data: np.ndarray) -> np.ndarray:
+        """Returns True for data that cannot be normalized"""
```

Comment:
I think the documentation does not explain the function well - since it rather return an array?